### ISSUE TITLE
`NoReturn` shouldn't be required in overload implementation

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2765]
The following code should not generate an error.

```python
from typing import NoReturn, overload

@overload
def func(x: int) -> NoReturn:
    ...

@overload
def func(x: str) -> str:
    ...

def func(x: int | str) -> str:
    ...
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.201, which I just published. It will also be included in the next release of pylance.
